apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/remote/EventReportServiceClient.java

```diff
@@ -0,0 +1,193 @@
+import org.apache.skywalking.apm.agent.core.logging.api.LogManager;
+import org.apache.skywalking.apm.network.common.v3.Commands;
+import org.apache.skywalking.apm.network.event.v3.Event;
+import org.apache.skywalking.apm.network.event.v3.EventServiceGrpc;
+import org.apache.skywalking.apm.network.event.v3.Source;
+import org.apache.skywalking.apm.network.event.v3.Type;
+
+import static org.apache.skywalking.apm.agent.core.remote.GRPCChannelStatus.CONNECTED;
+
+@DefaultImplementor
+public class EventReportServiceClient implements BootService, GRPCChannelListener {
+    private static final ILog LOGGER = LogManager.getLogger(EventReportServiceClient.class);
+
+    private final AtomicBoolean reported = new AtomicBoolean();
+
+    private Event.Builder startingEvent;
+
+    private EventServiceGrpc.EventServiceStub eventServiceStub;
+
+    private GRPCChannelStatus status;
+
+    private ServiceInstanceGenerator serviceInstanceGenerator;
+
+    @Override
+    public void prepare() throws Throwable {
+        ServiceManager.INSTANCE.findService(GRPCChannelManager.class).addChannelListener(this);
+
+        serviceInstanceGenerator = ServiceManager.INSTANCE.findService(ServiceInstanceGenerator.class);
+
+        if (serviceInstanceGenerator.isGenerated()) {
+            LOGGER.debug("The service instance is generated, no starting event will be reported");
+            return;
+        }
+
+        final RuntimeMXBean runtimeMxBean = ManagementFactory.getRuntimeMXBean();
+        startingEvent = Event.newBuilder()
+                             .setUuid(UUID.randomUUID().toString())
+                             .setName("Start")
+                             .setStartTime(runtimeMxBean.getStartTime())
+                             .setMessage("Start Java Application")
+                             .setType(Type.Normal)
+                             .setSource(
+                                 Source.newBuilder()
+                                       .setService(Config.Agent.SERVICE_NAME)
+                                       .setServiceInstance(Config.Agent.INSTANCE_NAME)
+                                       .build()
+                             )
+                             .putParameters(
+                                 "OPTS",
+                                 runtimeMxBean.getInputArguments()
+                                              .stream()
+                                              .sorted()
+                                              .collect(Collectors.joining(" "))
+                             );
+    }
+
+    @Override
+    public void boot() throws Throwable {
+
+    }
+
+    @Override
+    public void onComplete() throws Throwable {
+        reportStartingEvent();
+    }
+
+    @Override
+    public void shutdown() throws Throwable {
+        if (serviceInstanceGenerator.isGenerated()) {
+            // If the agent service instance name is randomly generated, ignore the shutdown signal.
```

Comment:
Do you mean, UUID-based instance doesn't report a shutdown event?

Reply:
Why? My feeling is, this is especially should have the event. When you see its metrics(such as CPM) suddenly goes down as 0, and find out the shut event exists, you know, this is not a performance accident.

Reply:
> Do you mean, UUID-based instance doesn't report a shutdown event?

Yes, UUID-based instance won't report start / shutdown events.

Reply:
> Why? My feeling is, this is especially should have the event. When you see its metrics(such as CPM) suddenly goes down as 0, and find out the shut event exists, you know, this is not a performance accident.

OK, this makes sense to me, I was overthinking about this